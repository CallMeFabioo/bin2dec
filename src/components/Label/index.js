import React from 'react';
import PropTypes from 'prop-types';

const Label = ({ labelFor, children, ...props }) => (
	<label
		className="block mb-2 text-sm text-3xl font-bold text-center text-gray-700"
		htmlFor={labelFor}
		{...props}
	>
		{children}
	</label>
);

Label.displayName = 'Label';

Label.propTypes = {
	labelFor: PropTypes.string,
	children: PropTypes.node,
};

export default Label;
