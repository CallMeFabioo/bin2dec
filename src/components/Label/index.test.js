import React from 'react';
import { render, screen } from '@testing-library/react';

import Label from '.';

it('renders a label', () => {
	const { container } = render(<Label />);

	expect(container).toMatchSnapshot();
});

it('renders a label passing custom props', () => {
	render(<Label data-testid="Label" />);

	expect(screen.queryByTestId(/Label/i)).toBeInTheDocument();
});
