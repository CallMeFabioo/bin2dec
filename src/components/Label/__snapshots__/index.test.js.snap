// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a label 1`] = `
<div>
  <label
    class="block mb-2 text-sm text-3xl font-bold text-center text-gray-700"
  />
</div>
`;
